service/controller/resource/tccpn/create.go

```diff
@@ -0,0 +1,515 @@
+	}
+
+	var iamPolicies *template.ParamsMainIAMPolicies
+	{
+		iamPolicies = &template.ParamsMainIAMPolicies{
+			ClusterID:        key.ClusterID(&cr),
+			EC2ServiceDomain: key.EC2ServiceDomain(cc.Status.TenantCluster.AWS.Region),
+			RegionARN:        key.RegionARN(cc.Status.TenantCluster.AWS.Region),
+			S3Bucket:         key.BucketName(&cr, cc.Status.TenantCluster.AWS.AccountID),
+		}
+
+		if encrypterBackend == encrypter.KMSBackend {
+			iamPolicies.KMSKeyARN = cc.Status.TenantCluster.Encryption.Key
+		}
+	}
+
+	return iamPolicies, nil
+}
+
+func newLaunchConfiguration(ctx context.Context, cr infrastructurev1alpha2.AWSControlPlane) (*template.ParamsMainLaunchConfiguration, error) {
+	cc, err := controllercontext.FromContext(ctx)
+	if err != nil {
+		return nil, microerror.Mask(err)
+	}
+
+	launchConfiguration := &template.ParamsMainLaunchConfiguration{
+		BlockDeviceMapping: template.ParamsMainLaunchConfigurationBlockDeviceMapping{
+			Docker: template.ParamsMainLaunchConfigurationBlockDeviceMappingDocker{
+				Volume: template.ParamsMainLaunchConfigurationBlockDeviceMappingDockerVolume{
+					Size: 100,
+				},
+			},
+			Kubelet: template.ParamsMainLaunchConfigurationBlockDeviceMappingKubelet{
+				Volume: template.ParamsMainLaunchConfigurationBlockDeviceMappingKubeletVolume{
+					Size: 100,
+				},
+			},
+			Logging: template.ParamsMainLaunchConfigurationBlockDeviceMappingLogging{
+				Volume: template.ParamsMainLaunchConfigurationBlockDeviceMappingLoggingVolume{
+					Size: 100,
+				},
+			},
+		},
+		Instance: template.ParamsMainLaunchConfigurationInstance{
+			Image:      key.ImageID(cc.Status.TenantCluster.AWS.Region),
+			Monitoring: true,
+			Type:       key.ControlPlaneInstanceType(cr),
+		},
+		SmallCloudConfig: template.ParamsMainLaunchConfigurationSmallCloudConfig{
+			S3URL: fmt.Sprintf("s3://%s/%s", key.BucketName(&cr, cc.Status.TenantCluster.AWS.AccountID), key.S3ObjectPathTCCPN(&cr)),
+		},
+	}
+
+	return launchConfiguration, nil
+}
+
+func newOutputs(ctx context.Context, cr infrastructurev1alpha2.AWSControlPlane) (*template.ParamsMainOutputs, error) {
+	outputs := &template.ParamsMainOutputs{
+		InstanceType:    key.ControlPlaneInstanceType(cr),
+		OperatorVersion: key.OperatorVersion(&cr),
+	}
+
+	return outputs, nil
+}
+
+func newSecurityGroupsParams(ctx context.Context, cr infrastructurev1alpha2.AWSControlPlane, apiWhiteList APIWhitelist) (*template.ParamsMainSecurityGroups, error) {
+	cc, err := controllercontext.FromContext(ctx)
+	if err != nil {
+		return nil, microerror.Mask(err)
+	}
+
+	var cfg securityConfig
+	{
+		cfg = securityConfig{
+			APIWhitelist:                    apiWhiteList,
+			ControlPlaneNATGatewayAddresses: cc.Status.ControlPlane.NATGateway.Addresses,
+			ControlPlaneVPCCidr:             cc.Status.ControlPlane.VPC.CIDR,
+			//TODO LH no idea if this is good or not .. was cluster.Status.Provider.Network.CIDR in tccp cluster
+			ProviderCIDR: cc.Status.ControlPlane.VPC.CIDR,
```

Comment:
We need to figure which resource is writing to `cluster.Status.Provider.Network.CIDR` and replicate that for the TCCPN stack.

Reply:
I am having trouble finding where this is written to. Only reference to it outside of the key is in a unittest helper (default_cluster.go) ? Is there a better way to find it ?

Reply:
You probably did not find more search results because the code is not 100%  consistent. When you drop `cluster.` from `cluster.Status.Provider.Network.CIDR` you figure out that the CIDR is set in the [IPAM cluster persister](https://github.com/giantswarm/aws-operator/blob/935fa07ced25b3d23395178b3ba71d8b1089ae09/service/controller/resource/ipam/cluster_persister.go#L45). This means the CIDR is stored in the `AWSCluster` CR status. Now, you work on the reconciliation of the `AWSControlPlane` CR and one approach would be to have a resource implementation that reads the CIDR from the other CR and puts it in the controller context.